dbt/include/firebolt/macros/materializations/models/incremental/incremental.sql

```
@@ -19,71 +19,80 @@
+           'Defaulting to append' % this, True) }}
+    {%- set strategy = 'append' -%}
+  {%- endif -%}
+  {# In following lines:
+
+     `target` is just a dbt `BaseRelation` object, not a DB table.
+
+     We're only retrieving `existing` to check for existence; `load_relation()`
+     returns a `BaseRelation` with the dictionary values of any relations that exist
+     in dbt's cache that share the identifier of the passed relation. If none
+     exist, returns None. Note that this does *not* create a table in the DB.
+
+     `target` is a relation with all the same fields as `existing`, but guaranteed
+     to be an actual `BaseRelation` object. #}
+  {%- set target = this.incorporate(type='table') -%}
+  {%- set existing = load_relation(this) -%}
+  {%- set new_records = make_temp_relation(target) -%}
   {{ drop_relation_if_exists(new_records) }}
-  {% set new_records = new_records.incorporate(type='table') %}
-  {% set on_schema_change = incremental_validate_on_schema_change(
+  {%- set new_records = new_records.incorporate(type='table') -%}
+  {%- set on_schema_change = incremental_validate_on_schema_change(
                                 config.get('on_schema_change'),
-                                default='ignore') %}
+                                default='ignore') -%}
 
   -- `BEGIN` happens here:
   {{ run_hooks(pre_hooks, inside_transaction=True) }}
-
+  {%- set partition_by = config.get('partition_by') -%}
   {# First check whether we want to full refresh for existing view or config reasons. #}
-  {% set do_full_refresh = (should_full_refresh() or existing.is_view) %}
+  {%- set do_full_refresh = (should_full_refresh()
+                            or existing.is_view
+                            or (strategy == 'insert_overwrite' and not partition_by)) %}
+  {%- if strategy == 'insert_overwrite' and not partition_by -%}
+    {{ log('`partition_by` is not specified for model %s. This '
```

I think you should raise an exception if the strategy is insert_overwrite and a partition isn't defined

---

@octavianzarzu30 might disagree, but I would prefer to get an error instead of only realizing there was an issue on the second run

---

Oh and I would make sure that it raises the exception at dbt compile time and not run time